File: apps/backend/internal/response/user.response.go
package response

type (
	// swagger:response UserResponse
	UserResponse struct {
		CommonModel
		Username      string `json:"username"` // username is uniqe of user in system
		Email         string `json:"email"`    // email feild for register and forgot password processed
		EmailVerifyed bool   `json:"email_verifyed"`
		FullName      string `json:"full_name"`
		RoleID        uint   `json:"role_id"`
	}
)

File: apps/www/src/types/user_response.ts
import { CommonModel } from "./common";
import { Image } from "./image_response";

export interface Role extends CommonModel {
  name: string;
}

export interface ProfileImage extends CommonModel {
  user_id: number;
  image_id: number;
  image: Image;
}

export interface UserResponse extends CommonModel {
  username: string;
  email: string;
  email_verifyed: boolean;
  full_name: string;
  role_id: number;
  profile_image?: ProfileImage[];
  role?: Role;
}
